apps/prairielearn/src/question-servers/freeform.js

```diff
@@ -1165,6 +1161,12 @@ async function renderPanel(panel, codeCaller, variant, submission, course, local
   data.options.client_files_question_url = locals.clientFilesQuestionUrl;
   data.options.client_files_course_url = locals.clientFilesCourseUrl;
   data.options.client_files_question_dynamic_url = locals.clientFilesQuestionGeneratedFileUrl;
+  data.options.course_element_files_url = assets
+    .courseElementAssetPath(course.commit_hash, locals.urlPrefix, '')
+    .replace(/\/+$/, '');
+  data.options.course_element_extension_files_url = assets
+    .courseElementExtensionAssetPath(course.commit_hash, locals.urlPrefix, '')
+    .replace(/\/+$/, '');
```

Comment:
This `replace()` stuff feels a bit nasty. Could we instead have `courseElementAssetPath()` accept a `null` (or `undefined`?) third argument and in that case return the path without the trailing slash?

Reply:
Very fair point! How does 1ca4ad3 look?

Reply:
Much cleaner! 🙏